core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/BoxImpl.java

```diff
@@ -42,12 +42,20 @@
   @ManyToAny(metaColumn = @Column(name = "targetType"))
   @MapKeyColumn(name = "position", unique = true)
   @MapKeyClass(String.class)
-  @JoinTable(name = "BoxPosition", joinColumns = { @JoinColumn(name = "targetId") }, inverseJoinColumns = {
-      @JoinColumn(name = "boxId") })
+  @JoinTable(name = "BoxPosition", joinColumns = { @JoinColumn(name = "boxId") }, inverseJoinColumns = {
+      @JoinColumn(name = "targetId") })
   @AnyMetaDef(idType = "long", metaType = "string", metaValues = {
       @MetaValue(targetEntity = LibraryImpl.class, value = "L"),
+      @MetaValue(targetEntity = DetailedLibraryImpl.class, value = "LD"),
       @MetaValue(targetEntity = PoolImpl.class, value = "P"),
-      @MetaValue(targetEntity = SampleImpl.class, value = "S")
+      @MetaValue(targetEntity = SampleImpl.class, value = "S"),
+      @MetaValue(targetEntity = SampleAliquotImpl.class, value = "SA"),
+      @MetaValue(targetEntity = IdentityImpl.class, value = "SI"),
```

Comment:
True, but we don't have any logic that prevents them from going in a box if they have a barcode. :-(
http://cdn2.stylecraze.com/wp-content/uploads/2013/11/Human-Inscription-with-Barcode-Tattoo-Design.jpg